ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/studio.py

```diff
@@ -40,6 +50,79 @@ class StudioMixin:
 
     studio_api_version: Literal["v1"] = "v1"
 
+    async def get_studio(
+        self: CVClient,
+        studio_id: str,
+        workspace_id: str,
+        time: datetime | None = None,
+        timeout: float = 10.0,
+    ) -> Studio:
+        """
+        Get Studio definition using arista.studio.v1.StudioService.GetOne.
+
+        The Studio GetOne API for the workspace does not return anything from mainline and does not return deletions in the workspace.
+        So to produce the Workspace Studio we need to fetch from the workspace, and if we find nothing, we need to check if the studio
+        got deleted in the workspace by retrieving config. Finally we can fetch from mainline.
+
+        Parameters:
+            studio_id: Unique identifier for the studio.
+            workspace_id: Unique identifier of the Workspace for which the information is fetched. Use "" for mainline.
+            time: Timestamp from which the information is fetched. `now()` if not set.
+            timeout: Timeout in seconds.
+
+        Returns:
+            Studio object.
+        """
+        request = StudioRequest(
+            # First attempt to fetch studio from workspace.
+            key=StudioKey(studio_id=studio_id, workspace_id=workspace_id),
+            time=time,
+        )
+        client = StudioServiceStub(self._channel)
+        try:
+            response = await client.get_one(request, metadata=self._metadata, timeout=timeout)
+            return response.value
+        except Exception as e:
+            e = get_cv_client_exception(e, f"Studio ID '{studio_id}, Workspace ID '{workspace_id}'") or e
+            if isinstance(e, CVResourceNotFound):
+                # Continue execution if we did not find any state in the workspace.
+                # This simply means the studio itself was not changed in this workspace.
+                pass
+            else:
+                raise e
+
+        # If we get here, it means no studio was returned by the workspace call.
+        # So now we fetch the studio config from the workspace to see if the studio was deleted in this workspace.
+        request = StudioConfigStreamRequest(
+            StudioConfig(
+                key=StudioKey(studio_id=studio_id, workspace_id=workspace_id),
+                remove=True,
+            ),
+            time=time,
+        )
+        client = StudioConfigServiceStub(self._channel)
+        try:
+            responses = client.get_all(request, metadata=self._metadata, timeout=timeout)
+            async for response in responses:
+                # If we get here it means we got an entry with "removed: True" so no need to look further.
+                raise CVResourceNotFound("The studio was deleted in the workspace.", f"Studio ID '{studio_id}, Workspace ID '{workspace_id}'")
```

Comment:
probably a bit nitpicking.

this piece of code will trigger the `except` line 110 - do we want this ? Or would we rather want to run the async for loop afterwards?
Feels like we are going to call `get_cv_client_exception` on an already crafted by ourselves exception otherwise

Reply:
Well get_cv_client_exception will ignore it and return none, so the `or e` will raise the original.

Reply:
true but this could be rewritten to avoid this double processing by putting the for loop outside of the try block